src/py-opentimelineio/opentimelineio/console/otiopluginfo.py

```diff
@@ -0,0 +1,250 @@
+        print("    implicit supported features:")
+        for feat in extra_features:
+            print("      {}".format(feat))
+
+
+def _schemadefs_formatted(feature_map, args):
+    print("    SchemaDefs:")
+    for sd in feature_map.keys():
+        print("      {}".format(sd))
+        _docs_formatted(feature_map[sd]['doc'], args)
+
+
+def _docs_formatted(docstring, arg_map, indent=4):
+    long_docs = arg_map.get('long_docs')
+
+    if long_docs:
+        prefix = " " * indent + "doc (long): "
+    else:
+        prefix = " " * indent + "doc (short): "
+
+    initial_indent = prefix
+    subsequent_indent = " " * len(prefix)
+
+    block = docstring.split("\n")
+    fmt_block = []
+    for line in block:
+        line = textwrap.fill(
+            line,
+            initial_indent=initial_indent,
+            subsequent_indent=subsequent_indent,
+            width=len(subsequent_indent) + 80,
+        )
+        initial_indent = subsequent_indent
+        fmt_block.append(line)
+
+    if long_docs:
+        text = "\n".join(fmt_block)
+    else:
+        text = fmt_block[0]
+
+    print(text)
+
+
+_FORMATTER = {
+    "supported features": _supported_features_formatted,
+    "SchemaDefs": _schemadefs_formatted,
+    "doc": _docs_formatted,
+}
+
+_FIELDS_TO_SKIP = frozenset(["name"])
+
+
+def _print_field(key, val, **args):
+    # if attribute doesn't hit any of the user specified patterns
+    if (
+        not any(fnmatch.filter([key], pt) for pt in args['attribs'])
+        or key in _FIELDS_TO_SKIP
+    ):
+        return
+
+    if key in _FORMATTER:
+        _FORMATTER[key](val, args)
+        return
+
+    print("    {}: {}".format(key, val))
+
+
+def main():
+    """  main entry point  """
+    args = _parsed_args()
+
+    plugin_types = args.plugin_types
+
+    if 'all' in plugin_types:
+        # all means the full list of built in plugins
+        plugin_types = otio.plugins.manifest.OTIO_PLUGIN_TYPES
+
+    # load all the otio plugins
+    active_plugin_manifest = otio.plugins.ActiveManifest()
+
+    # list the loaded manifests
+    print("Manifests loaded:")
+    for mf in active_plugin_manifest.source_files:
+        print("  {}".format(mf))
+
+    for pt in plugin_types:
+        # hooks have special code (see below)
+        if pt in ["hooks"]:
+            continue
+
+        # header
+        print("")
+        print("{}:".format(pt))
+
+        # filter plugins by the patterns passed in on the command line
+        plugin_by_type = getattr(active_plugin_manifest, pt)
+        plugins = [
+            p for p in plugin_by_type
+            if fnmatch.filter([p.name], args.plugpattern)
+        ]
+
+        # if nothing is found of that type that matches the filter
+        if not plugins:
+            print("    (none found)")
+
+        for plug in plugins:
+            print("  {}".format(plug.name))
+
+            try:
+                info = plug.plugin_info_map()
+            except Exception, err:
```

Comment:
I have gotten in the habit of using "as" here: 

https://stackoverflow.com/questions/2535760/python-try-except-comma-vs-as-in-except